### PR TITLE
Try to use exchanges during database replicated replica recovery process

### PR DESCRIPTION
**Pull request by nikitamikhaylov**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in official stable or prestable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
During recovering of the lost replica there could a situation where we need to atomically swap two table names (use EXCHANGE), but instead previously we tried to use two RENAME queries. Which was obviously failed and moreover failed the whole recovery process of the database replica. 

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Databases/DatabaseReplicated.cpp [hunk 1]
```diff
@@ -719,8 +730,9 @@ void DatabaseReplicated::recoverLostReplica(const ZooKeeperPtr & current_zookeep
             {
                 if (name != it->second)
                 {
+                    String intermediate_name = fmt::format("{}-{}", name, sipHash64(fmt::format("{}-{}", name, salt)));
```
Comment (tavplubix) [inline src/Databases/DatabaseReplicated.cpp:733]:
```suggestion
                    String intermediate_name = fmt::format(".rename-{}-{}", name, sipHash64(fmt::format("{}-{}", name, salt)));
```

Comment (tavplubix) [reply src/Databases/DatabaseReplicated.cpp:733]: Also we can skip renaming to `intermediate` name if `from` name already starts with `".rename-"` and `to` name does not. Imagine that for some reason we constantly fail to do the second stage of renaming. We will append some random number on each iteration until we end up with extremely long table names.

File: src/Databases/DatabaseReplicated.cpp [hunk 2]
```diff
@@ -719,8 +730,15 @@ void DatabaseReplicated::recoverLostReplica(const ZooKeeperPtr & current_zookeep
             {
                 if (name != it->second)
                 {
+                    String intermediate_name;
+                    /// Possibly we failed to rename it on previous iteration
+                    /// And this table was already renamed to an intermediate name
+                    if (startsWith(name, ".rename-"))
```
Comment (tavplubix) [inline src/Databases/DatabaseReplicated.cpp:736]:
```suggestion
                    if (startsWith(name, ".rename-") && !startsWith(it->second, ".rename-"))
```

**Review by tavplubix — APPROVED** [review]

**Comment by nikitamikhaylov** [comment]
Integration test failure for `test_distributed_queries_stress/test.py` unrelated and I don't know the reason why it failed `Stdout:OCI runtime exec failed: exec failed: unable to start container process: error executing setns process: exit status 1: unknown (cluster.py:103, run_and_check)`

**Pull request merged**